### ISSUE TITLE
Directives in lintr config file ignored?

### ISSUE THREAD

**Issue by IndrajeetPatil** [r-lib/lintr#2309]
E.g., in the package below, `.lintr` turns off `implicit_integer_linter`, and yet `lintr::lint_package()` flags those lints. 

``` r
dir <- tempfile()
pkg <- usethis::create_package(dir, open = FALSE, rstudio = FALSE)
setwd(pkg)
fs::dir_create("R")
writeLines(
  c("x <- 1", "NULL"), 
  "R/script.R"
)
writeLines(
  c("linters: linters_with_defaults(implicit_integer_linter = NULL, defaults = linters_with_tags(tags = NULL))"), 
  ".lintr"
)
devtools::install(quiet = TRUE)

lintr::lint_package(linters = lintr::all_linters())
#> R/script.R:1:7: style: [implicit_integer_linter] Integers should not be implicit. Use the form 1L for integers or 1.0 for doubles.
#> x <- 1
#>      ~^
```

<sup>Created on 2023-11-18 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

Is this a bug, or do I harbour an incorrect mental model of how the config file is being used?

**Comment by IndrajeetPatil** [comment]
Not clear if this is also related to https://github.com/r-lib/lintr/issues/2296

**Comment by AshesITR** [comment]
Not sure if that's the issue, but you don't need to nest.

```r
linters_with_tags(tags = NULL, implicit_integer_linter = NULL)
```

Would be a simpler config with the same effect.

**Comment by AshesITR** [comment]
Oh wait, you are overwriting `linters=` via a direct argument to `lint_package()`. This is expected behavior.
